source/common/router/header_parser.h

```diff
@@ -0,0 +1,63 @@
+#pragma once
+
+#include <list>
+#include <string>
+
+#include "envoy/access_log/access_log.h"
+#include "envoy/http/header_map.h"
+
+#include "common/protobuf/protobuf.h"
+#include "common/router/header_formatter.h"
+
+#include "api/base.pb.h"
+
+namespace Envoy {
+namespace Router {
+
+class RequestHeaderParser;
+typedef std::unique_ptr<RequestHeaderParser> RequestHeaderParserPtr;
+
+/**
+ * This class provides request-time generation of upstream request headers. Header configurations
+ * are pre-parsed to select between constant values and values based on the evaluation of
+ * AccessLog::RequestInfo fields.
+ */
+class RequestHeaderParser {
```

Comment:
Could RequestHeaderParser and ResponseHeaderParser be combined?  It looks like they do almost exactly the same thing.

Reply:
They can be, but I don't want to do that until the PR that introduces support for RequestInfo lookups for response headers.